src/dmd/expressionsem.d

```diff
@@ -3740,7 +3767,7 @@ private extern (C++) final class ExpressionSemanticVisitor : Visitor
         }
         else
         {
-            exp.error("new can only create structs, dynamic arrays or class objects, not `%s`'s", exp.type.toChars());
+            exp.error("cannot create a `%s` with `new`", exp.type.toChars());
```

Comment:
The old error message was more appropriate because it explained the restrictions of `new`

Reply:
The old message missed some allowable types as @thewilsonator pointed out... Seems like an exhaustive list is a bit overwhelming, but I don't have a strong opinion either way

Reply:
Fair enough. Maybe something like this would better but no strong opinion either.

```suggestion
            exp.error("`new`cannot create a `%s`", exp.type.toChars());
```